ballet/eng/misc.py

```diff
@@ -1,18 +1,43 @@
 import funcy
 import numpy as np
 import pandas as pd
+from scipy.special import boxcox1p
+from scipy.stats import skew
 
 from ballet.eng.base import BaseTransformer, SimpleFunctionTransformer
 from ballet.util import get_arr_desc
 
-__all__ = ['IdentityTransformer', 'ValueReplacer', 'NamedFramer']
+__all__ = ['IdentityTransformer', 'BoxCoxTransformer', 'ValueReplacer', 'NamedFramer']
 
 
 class IdentityTransformer(SimpleFunctionTransformer):
     def __init__(self):
         super().__init__(funcy.identity)
 
 
+class BoxCoxTransformer(BaseTransformer):
+    def __init__(self, threshold, lmbda=0):
+        super().__init__()
+        self.threshold = threshold
+        self.lmbda = lmbda
+
+    def fit(self, X, y=None, **fit_args):
+        self.features_to_transform_ = abs(skew(X)) > self.threshold
+        if (isinstance(X, pd.DataFrame)):
+            # Hack to get a mask over columns
+            self.features_to_transform_ = X.columns.putmask(~self.features_to_transform_, None) & X.columns
+        return self
+
+    def transform(self, X, **transform_args):
+        if isinstance(X, pd.DataFrame):
+            return boxcox1p(X[self.features_to_transform_], self.lmbda)
+        elif isinstance(X, pd.Series) and self.features_to_transform_:
+            return boxcox1p(X, self.lmbda)
+        elif isinstance(X, np.ndarray):
+            return boxcox1p(X[:, self.features_to_transform_], self.lmbda)
```

Comment:
What if `X` is not an instance of these three datatypes?
- if `self.features_to_transform_` is truthy, then raise an error as we don't know how to select out the features.
- if `self.features_to_transform_` is falsy, they we can pass through `X` directly